src/de/jost_net/JVerein/io/AbrechnungSEPA.java

```diff
@@ -826,7 +833,10 @@ private void writeMitgliedskonto(Object mitglied, Date datum, String zweck1,
       mk = (Mitgliedskonto) Einstellungen.getDBService()
           .createObject(Mitgliedskonto.class, null);
       mk.setAbrechnungslauf(abrl);
-      mk.setZahlungsweg(mg.getZahlungsweg());
+      if(zahlungsweg != null)
+        mk.setZahlungsweg(zahlungsweg);
+      else
+        mk.setZahlungsweg(mg.getZahlungsweg());
```

Comment:
Müsste hier bei Vollzahler nicht der Zahlungsweg von Vollzahler geholt werden?

Reply:
Bisher werden die Sollbuchungen auf des Mitglied und nicht auf den Vollzahler ausgestellt. Das müsste geändert werden. Dann sollte zum Zahlungsgrund der Sollbuchung der Name des Mitglieds hinzugefügt werden, sonst lässt sich nicht erkennen für wen die Sollbuchung ist.

Wenn wir es so machen, dass auch bei abweichendem Zahlungsweg immer der Vollzahler bezahlt ist es nicht möglich dem Mitglied eine Zahlung zuzuordnen die er selbst bezahlt. Beim anderen vorgehen könnte man das durch den abweihenden Zahlungsweg machen.

Reply:
Nein, die Sollbuchungen sollen schon auf das Mitglied ausgestellt bleiben.

Aber wenn die Sollbuchung beim Familienverband vom vollzahlendem Familienmitglied bezahlt wird geht das ja über die Zahlungsart die beim Vollzahler konfiguriert ist und nicht mit der die beim Mitglied gesetzt ist.

Reply:
Ich glaube das ist gelöst wenn man in Zeile 455 den Zahlungsweg des mZahler übergibt statt null.

Reply:
Ja, stimmt so mache ich es.

Bei Lastschrift ist so alles gut. Bei Überweisung ist jedoch die Sollbuchung auf das Mitglied und man kann nirgends sehen, dass sie vom Vollzahler gezahlt werden sollt. Da müsste man dann auch bei der automatischen Zuordnung der Buchungen zum Mitglied den Test erweitern.

Reply:
Das Problem ist eigentlich, dass man an der Sollbuchung nicht sehen kann wer der Zahler ist. Das ist das gleiche Problem wie bei den Spendenbescheinigungen.
Vielleicht könnte man das zusammen löschen. Anders als ich vorgeschlagen hatte könnte man in der Sollbuchung ein Attribut "zahler" einführen. Den könnte man beim Abrechnungslauf setzen. Dann könnte man bei der Zuordnung der Buchung auf den Zahler testen. Und beim Erstellen der Spendenbescheinigung den Zahler verwenden. Das wäre auch die einfachste Lösung für das Problem der Spendenbescheinigung. 
Es löst nur nicht das Problem wenn ein abweichender Kontoinhaber eingetragen ist. Da bleibt das Mitglied als Zahler weil es den alternativen Kontoinhaber nicht als (Nicht-)Mitglied gibt. Da muss man dann die Spendenbescheinigung per Hand korrigieren oder man benutzt das Familienkonzept. Beim Familienkonzept müsste man dann aber auch Nicht-Mitglieder als Vollzahler erlauben.

Ich denke das könnte man in einem extra PR machen.

Reply:
finde ich eine gute Idee, können wir ja in #467 weiter diskutieren